shepherd_bms/src/main.cpp

```diff
@@ -72,11 +72,32 @@ void balanceCells(AccumulatorData_t *bms_data)
  */
 bool balancingCheck(AccumulatorData_t *bmsdata)
 {
+	static Timer balanceTimer;
+
 	if (!compute.isCharging()) return false;
 	if (bmsdata->maxTemp.val > MAX_CELL_TEMP_BAL) return false;
 	if (bmsdata->maxVoltage.val <= (BAL_MIN_V * 10000)) return false;
 	if(bmsdata->deltVoltage <= (MAX_DELTA_V * 10000)) return false;
 
+	//balance timer state 0 == not balancing
+	//balance timer state 1 == balancing
+
+	if (balanceTimer.isTimerExpired() && balanceTimerState == 0)
```

Comment:
nit - either make it a boolean ( ```if (isBalancing)```  ) or enumerate these states ( ``` if (balanceTimerState == NOT_BALANCING)``` ) or something, try and avoid magic numbers where possible